pkg/controller/deployment/deployment_controller.go

```diff
@@ -367,24 +367,26 @@ func (dc *DeploymentController) deletePod(logger klog.Logger, obj interface{}) {
 			return
 		}
 	}
+	d := dc.getDeploymentForPod(logger, pod)
+	if d == nil || d.Spec.Strategy.Type != apps.RecreateDeploymentStrategyType {
```

Comment:
`d.Spec.Strategy.Type != apps.RecreateDeploymentStrategyType` could be checked after the log line.

Reply:
Updated.